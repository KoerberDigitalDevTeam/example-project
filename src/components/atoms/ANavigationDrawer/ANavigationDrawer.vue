<template>
  <v-navigation-drawer
    v-testId.navigation-drawer
    :permanent="!$vuetify.breakpoint.mobile"
    touchless
    class="a-navigation-drawer"
    v-bind="[$attrs, $props]"
    v-on="$listeners"
  >
    <slot />
  </v-navigation-drawer>
</template>

<script>
export default {
  name: 'ANavigationDrawer',

  props: {
    miniVariant: {
      type: Boolean,
      default: false,
    },
  },
};
</script>
