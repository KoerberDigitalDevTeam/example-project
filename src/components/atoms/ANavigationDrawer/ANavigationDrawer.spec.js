import { shallowMount } from '@vue/test-utils';
import Vuetify from 'vuetify';
import ANavigationDrawer from '@/components/atoms/ANavigationDrawer/ANavigationDrawer.vue';

function createWrapper(options = {}) {
  return shallowMount(ANavigationDrawer, {
    vuetify: new Vuetify(),
    ...options,
  });
}

describe('ANavigationDrawer', () => {
  describe('Template:', () => {
    it('Should render an ANavigationDrawer', () => {
      const wrapper = createWrapper();

      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('Props Interface:', () => {
    it('Should render an ANavigationDrawer with miniVariant prop', () => {
      const wrapper = createWrapper({
        miniVariant: false,
      });

      expect(wrapper.vm.$props.miniVariant).toBe(false);
    });
  });

  it('Should have miniVariant prop and must be Boolean', () => {
    expect(ANavigationDrawer.props).toMatchObject({
      miniVariant: {
        type: Boolean,
        default: false,
      },
    });
  });
});
