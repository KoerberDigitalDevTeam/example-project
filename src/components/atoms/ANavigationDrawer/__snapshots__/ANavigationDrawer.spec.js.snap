// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ANavigationDrawer Template: Should render an ANavigationDrawer 1`] = `<v-navigation-drawer-stub mobilebreakpoint="1264" height="100%" minivariantwidth="56" src="" tag="aside" touchless="true" width="256" class="a-navigation-drawer" data-testid="navigation-drawer"></v-navigation-drawer-stub>`;
